tests/test_queuing.py

```diff
@@ -283,3 +283,39 @@ def template(x):
             template(3)
 
         assert str(recorder) == expected_output
+
+
+class AnnotatingTensor(qml.operation.Tensor):
+    """Dummy tensor class that queues itself on initialization
+    to an annotating queue."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.queue()
+
+    def queue(self):
+        qml.QueuingContext.append(self)
+
+        for o in self.obs:
+            try:
+                qml.QueuingContext.active_context().update_info(o, owner=self)
```

Comment:
Nice, I suppose this will get removed once the `Tensor` class itself has this behaviour?

Reply:
Maybe since this doesn't really do anything related to `Tensor`, it can just be a generic dummy object that registers itself as an owner of some other ops?

Reply:
> Nice, I suppose this will get removed once the Tensor class itself has this behaviour?

Yep, just wanted to test that annotations work before integration happens!

Question: I keep thinking that

```python
qml.QueuingContext.active_context().update_info(o, owner=self)
```
is pretty ugly. We could upgrade `update_info` and `get_info` to top-level methods on `QueuingContext`, and just have them raise `NotImplementedError` if the particular subclass doesn't support them.